src/agent/onefuzz-agent/src/tasks/report/crash_report.rs

```diff
@@ -72,7 +72,7 @@ async fn upload_deduped(report: &CrashReport, container: &BlobContainerUrl) -> R
         .header("If-None-Match", "*")
         .send_retry_default()
         .await?;
-    if result.status() != StatusCode::NOT_MODIFIED {
```

Comment:
The expected return code here would have been a [409](https://docs.microsoft.com/en-us/rest/api/storageservices/blob-service-error-codes) with a "BlobAlreadyExists" in the response headers. 
But i just just simplified the logic to just check that the blob was created instead.

Reply:
Hmm, I think that's inconsistent with the docs I link to above. Or at least, it's not clear what behavior to expect, given our API version.

This raises a question: are we using a constant REST API version? It looks like we aren't.

With a SAS URL, if the API version is not otherwise specified, then the API version used is derived from [the version used in signature](
https://docs.microsoft.com/en-us/rest/api/storageservices/versioning-for-the-azure-storage-services#requests-authorized-with-a-shared-access-signature).

It looks like we don't set the API version in the `BlobClient` methods, and we visibly are not doing so here, so I'm wondering if the API changed and this code stopped working correctly for that reason. (Either way, your new check is better).

Reply:
Opened #424 to capture the above concerns.